### PR TITLE
Update Test suite

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---

- Updates Python version used in test scripts to `3.7` from `3.6`
- Updates the base images to versions of the OS that we actually support (thanks @bcambl  for the Centos stuff)
- Adds two new tests to prove the OS compatibility check is working/falls over on an invalid OS

### CODE REVIEW

File: test/test_automated_install.py
Comment (pvogt09) [inline test/test_automated_install.py:492]:
```suggestion
def test_os_check_passes(Pihole):
```

**Comment by pvogt09** [comment]
I tried something similiar in https://github.com/pvogt09/pi-hole/tree/feature/test_install_deps, but the installation of the web dependencies would never pass the test because of missing/wrong `php` versions in the images. In case you find anything useful therein, feel free to use it.

This PR will also makes #3213 obsolete.

I also have a branch with all supported OS (at the time of the creation of that branch) in the test cases, in case that should be needed in https://github.com/pvogt09/pi-hole/tree/feature/test_all_supported. I have to admit that this increases the time for the travis build to sometimes more than 15 minutes and might not be desired.

**Comment by PromoFaux** [comment]
I've never really played about with the tests before, it was only yesterday where I figured we should probably have a test for the new functionality of the supported version checker that I had a look!

I hit a bit of a brick wall, though. I'd like to try and get a test that mocks the output of PRETTY_NAME/Version so that we can test that the os check function will fail in case of an unsupported OS. But my head hurts

File: test/test_automated_install.py
Comment (dschaper) [inline test/test_automated_install.py:501]:
```suggestion
    source /opt/pihole/basic-install.sh
    pretty_name = 'PRETTY_NAME="Debian GNU/Linux 10 (buster)"'
    with open('/etc/os-release','w') as f:
        f.write(pretty_name)
    os_check
```

Or something in that same vein.

Comment (dschaper) [reply test/test_automated_install.py:501]: Might have to break out the python code to a line before `.run`, I don't remember if you can do python inside the `run` or not.

Comment (PromoFaux) [reply test/test_automated_install.py:501]: The only thing was the file `*-release` file having different names in different distros...

Comment (dschaper) [reply test/test_automated_install.py:501]: Okay, just change the name of the file you're writing to when you change the `pretty_name` string.

**Comment by PromoFaux** [comment]
OK, I think this is now ready for a re-review. 

As noted by @pvogt09, this will render #3213 obsolete. One thing I've noted from that one, however, is that the target is `stretch`, and here I have targeted `buster`. Someone else can make the decision which one we use, it's all the same to me :), provided it's not `jessie` or `wheezy` ;)

**Comment by pvogt09** [comment]
I chose the oldest supported version in #3213. As mentioned there this seems legit, but I would vote for testing all supported versions. I can set up a PR for this as soon as this one is merged, since I already have done something in this direction.

**Comment by dschaper** [comment]
I prefer lowest supported if we only have one version. My thinking is newer versions would still support the packages in the previous version, it's very rare for an OS version release to have changes that diverge it from a previous version.

**Comment by bcambl** [comment]
> 
> 
> I prefer lowest supported if we only have one version. My thinking is newer versions would still support the packages in the previous version, it's very rare for an OS version release to have changes that diverge it from a previous version.

CentOS falls into this catagory. CentOS8 is very similar to Fedora (new php, dnf package manager, etc).  
CentOS7 ships with an old PHP version and requires the additional image for PHP logic testing. Once CentOS7 support is dropped, then we can go back to a single container images for both `centos` and `fedora`.

**Review by Adam-SAI — DISMISSED** [review]

**Review by diginc — DISMISSED** [review]

**Comment by PromoFaux** [comment]
Sorry, @diginc, had to dismiss your review to make a change.

> I can set up a PR for this as soon as this one is merged, since I already have done something in this direction.

Sounds good to me.

**Review by diginc — APPROVED** [review]

**Pull request merged**
